sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/filter/AlwaysFalse.java

```diff
@@ -47,4 +47,9 @@ public int hashCode() {
 
   @Override
   public NamedReference[] references() { return EMPTY_REFERENCE; }
+
+  @Override
+  public org.apache.spark.sql.sources.Filter toV1() {
```

Comment:
Do we expect data source developers to use it? I thought we will add internal utils to convert between v1 and v2 filters, if it's for internal usage only.

Reply:
It's for internal use only. I will remove this and add a method in `DataSourceUtils` to do the conversion.